asllib/doc/ASLSemanticsReference.tex

```diff
@@ -291,107 +2073,73 @@ \section{SemanticsRule.EGlobalVar \label{sec:SemanticsRule.EGlobalVar}}
 
-%  \subsection{Example: SemanticsRuleEGlobalVarError.asl}
-%    The program:
-%    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.EGlobalVarError.asl}
-%    raises a type error because \texttt{x} is already declared locally in
-%    \texttt{stdlib.asl}.
-
   \subsection{Code}
   \VerbatimInput[firstline=\EGlobalVarBegin, lastline=\EGlobalVarEnd]{../Interpreter.ml}
 
 \begin{formal}
-  \subsection{Formally: sequential case}
-  \begin{align}
-  \interp{\texttt{x}}(\texttt{env}) & \triangleq \left\{ (\texttt{env}[\texttt{x}], \texttt{env})) \right\}
-  \label{eq:sem-seq-var}
-  \end{align}
-
-  \subsection{Formally: concurrent case}
-  \begin{align}
-  \interp{\texttt{x}}(\texttt{env}) & \triangleq \left\{ (\texttt{env}[\texttt{x}], \texttt{env}, \texttt{R(x)}) \right\}
-  \label{eq:sem-conc-var}
-  \end{align}
+  \subsection{Formally}
+Due to how $\denv[\cdot]$ distinguishes between local variables and global variables,
+we can use the same rule as for local variable expressions:
+  \begin{mathpar}
+  \inferrule{
+    \env=(\tenv, \denv)\\
+    \vv = \denv[\vx] \\ \vg = \ReadEffect(\vx)}
+  {\evalexpr{\denv, \evar(\vx)} \evalarrow \Normal((\vv, \vg), \env)}
+  \end{mathpar}
 \end{formal}
 
 \isempty{\subsection{Comments}}
 
-\section{SemanticsRule.EUndefIdent \label{sec:SemanticsRule.EUndefIdent}}
-
-  \subsection{Prose}
-  Evaluation of the expression \texttt{e} under environment \texttt{env} is
-  \texttt{res} and all of the following apply:
-  \begin{itemize}
-  \item \texttt{e} denotes a variable \texttt{x} which is not bound in \texttt{env};
-  \item \texttt{res} is an ``\texttt{Undefined Identifier}'' error.
-  \end{itemize}
-
-  \subsection{Example: SemanticsRule.EUndefIdent.asl}
-    The program:
-    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.EUndefIdent.asl}
-    raises an \texttt{Undefined Identifier} error: the variable \texttt{y} is
-    undefined in the environment where $\mathtt{x}$ is bound to $\mathtt{42}$.
-
-  \subsection{Code}
-  \VerbatimInput[firstline=\EUndefIdentBegin, lastline=\EUndefIdentEnd]{../Interpreter.ml}
-
-\begin{emptyformal}
-  \subsection{Formally: sequential case}
-
-  \subsection{Formally: concurrent case}
-\end{emptyformal}
-
-\isempty{\subsection{Comments}}
-
 \section{SemanticsRule.BinopAnd \label{sec:SemanticsRule.BinopAnd}}
 
   \subsection{Prose}
-  Evaluation of the expression \texttt{e} under environment \texttt{env} is
-  \texttt{(res,new\_env)} and all of the following apply:
-  \begin{itemize}
-  \item \texttt{e} denotes a conjunction over two expressions \texttt{e1} and \texttt{e2};
-  \item \texttt{(res,new\_env)} is the result of the evaluation of \texttt{if
-e1 then e2 else false} (see Section~\ref{sec:SemanticsRule.ECond}).
-  \end{itemize}
+  An expression \texttt{e}, which denotes a logical conjunction of two expressions \texttt{e1} and \texttt{e2},
+  is evaluated as \texttt{if e1 then e2 else false} (see Section~\ref{sec:SemanticsRule.ECond}).
```

Comment:
why have you changed the transliteration here? please could you avoid such changes? a lot of previous work went into making the format of the prose paragraphs consistent throughout the document - please consider reverting those changes, here, above and throughout. Essentially the prose should only change if there is an error - the formatting is here for consistency.